src/main/java/christmas/domain/menu/OrderParser.java

```diff
@@ -0,0 +1,55 @@
+package christmas.domain.menu;
+
+import christmas.message.ExceptionMessage;
+import java.util.Arrays;
+import java.util.EnumMap;
+import java.util.Map;
+
+public class OrderParser {
+    private static final int MAX_MENU_COUNT = 20;
+    private static final String DELIMITER_MENU_ORDER = ",";
+
+    private OrderParser() {
+        // 불필요한 인스턴스 생성 방지
+    }
+
+    public static Map<Menu, Integer> parse(String inputValue) {
+        Map<Menu, Integer> orderRepository = new EnumMap<>(Menu.class);
+        Arrays.stream(inputValue.split(DELIMITER_MENU_ORDER))
+                        .forEach(menuString -> addMenuToOrderRepository(orderRepository, menuString));
+        validateOnlyDrink(orderRepository);
+        validateMaxMenuCount(orderRepository);
+        return orderRepository;
+    }
+
+    private static void addMenuToOrderRepository(Map<Menu, Integer> orderRepository, String menuString) {
+        Order order = Order.createByString(menuString);
+        validateDuplicateMenu(orderRepository, order.menu());
+        orderRepository.put(order.menu(), order.count());
+    }
+
+    private static void validateDuplicateMenu(Map<Menu, Integer> result, Menu menu) {
+        if (result.containsKey(menu)) {
+            throw new IllegalArgumentException(ExceptionMessage.INVALID_ORDER);
+        }
+    }
+
+    private static void validateOnlyDrink(Map<Menu, Integer> result) {
+        int drinkMenuCount = (int) result.keySet()
+                .stream()
+                .filter(menu -> menu.isSameType(FoodType.DRINK))
+                .count();
+        if (drinkMenuCount == result.size()) {
+            throw new IllegalArgumentException(ExceptionMessage.INVALID_ORDER);
+        }
```

Comment:
모두 음료수인지 확인하는 스트림 체인 로직을 리드미 공장장 해빈님 코드에서 훔쳐왔어요....!
타칸님 코드에 맞게 수정해 보았습니다 ^^
```suggestion
        boolean isAllDrink = result.keySet()
                .stream()
                .allMatch(menu -> menu.isSameType(FoodType.DRINK));
        if(isAllDrink){
            throw new IllegalArgumentException(ExceptionMessage.INVALID_ORDER);
        }
```

Reply:
@zangsu `allMatch` 를 활용하면 되군요! 감사합니다! 아직 스트림에 대해선 배울게 많은것 같네요! ㅎㅎ